src/controller/CHIPDeviceController.cpp

```diff
@@ -139,6 +139,10 @@ CHIP_ERROR DeviceController::Init(ControllerInitParams params)
         }
     }
 
+#if defined(MATTER_CUSTOM_TRACE) && MATTER_CUSTOM_TRACE
+    chip::InitializeTracing();
```

Comment:
This will get called multiple times when multiple controllers are brought up, or not at all for a server-only device.  Why is this the right place to call this?

Reply:
I guess I might move this to more specific place. My use case is for android application, does it make more sense to move it to CHIPDeviceController-JNI.cpp JNI_OnLoad?

Reply:
I don't know.  All I can tell you is that if this method is not idempotent it needs to be some place where it will only be called once.  Where in the Android app is reasonable is a question for the folks familiar with that app...